src/modules/core/components/ColorSelect/ColorSelect.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useCallback, useMemo, ReactNode, ComponentProps } from 'react';
+import { defineMessages } from 'react-intl';
+
+import { Select, SelectOption, Form } from '~core/Fields';
+import ColorTag, { Color } from '~core/ColorTag';
+
+import styles from './ColorSelect.css';
+
+const MSG = defineMessages({
+  labelColorSelect: {
+    id: 'ColorSelect.labelColorSelect',
+    defaultMessage: 'Select color',
+  },
+});
+
+interface FormValues {
+  activeColor: Color;
+}
+
+interface Props {
+  /** Should `select` be disabled */
+  disabled?: boolean;
+
+  /** Active color */
+  activeOption: Color;
+
+  /** Callback function, called after value is changed */
+  onColorChange?: (color: Color) => any;
+}
+
+const displayName = 'ColorSelect';
+
+const ColorSelect = ({ disabled, activeOption, onColorChange }: Props) => {
+  const onChange = useCallback(
+    (color: Color) => {
+      if (onColorChange) {
+        return onColorChange(color);
+      }
+      return null;
+    },
+    [onColorChange],
+  );
+
+  const renderActiveOption = useCallback<
+    (option: SelectOption | undefined, label: string) => ReactNode
+  >(() => {
+    return <ColorTag color={activeOption || Color.Yellow} />;
+  }, [activeOption]);
+
+  const options = useMemo<ComponentProps<typeof Select>['options']>(() => {
+    const colors = Object.keys(Color).filter(
+      (val) => typeof Color[val as any] === 'number',
+    );
+    return [
+      ...colors.map((color) => {
+        return {
+          children: <ColorTag color={Color[color]} />,
+          label: `${color}`,
+          value: `${Color[color]}`,
+        };
+      }),
+    ];
+  }, []);
+
+  return (
+    <Form<FormValues>
+      initialValues={{
+        activeColor: activeOption || Color.Yellow,
+      }}
+      onSubmit={() => {}}
+    >
+      <div className={styles.wrapper}>
+        <Select
+          appearance={{
+            theme: 'grid',
+          }}
+          elementOnly
+          label={MSG.labelColorSelect}
+          name="activeColor"
+          onChange={(val) => {
+            onChange(Number(val));
+          }}
+          options={options}
```

Comment:
When re-opening the dropdown, the active color should have a blue outline, when shown along the other colors in the grid:

![Screenshot from 2020-12-09 20-27-08](https://user-images.githubusercontent.com/1193222/101671261-ef3bad80-3a5c-11eb-85e8-5926970623d2.png)

vs. the current one:

![Screenshot from 2020-12-09 20-19-13](https://user-images.githubusercontent.com/1193222/101671281-f82c7f00-3a5c-11eb-935c-2db43b9f8984.png)

_(notice that no color is "selected" in the dropdown)_

Reply:
Also, please ask _product_ if the "selected" color in the dropdown should have that white dot in the middle:

![Screenshot from 2020-12-09 20-18-41](https://user-images.githubusercontent.com/1193222/101671384-172b1100-3a5d-11eb-82f9-150846ad4cd6.png)

I personally think it's an error in the figma file, but please check.